### ISSUE TITLE
Panic in from_little_endian

### ISSUE THREAD

**Issue by That3Percent** [paritytech/parity-common#507]
I would expect the following round trip to work:
```rust
let mut b32 = <[u8; 32]>::default();
let a = U256::from(10);
a.to_little_endian(&mut b32);
let b = U256::from_little_endian(&b32[..]);
assert_eq!(a, b);
```

But instead it panics with:
```
'called `Result::unwrap()` on an `Err` value: TryFromSliceError(())'`
```

**Comment by ordian** [comment]
Thanks for the report @That3Percent.

I'm not able to reproduce it on my little-endian CPU, are you running it on a big-endian one?
The error is likely coming `LittleEndian::read_u64`, which is called in `U256::from_little_endian`, but I don't understand how is that possible.

Could you provide more details about the environment you're running in and the version of `uint`? Could does it reproduce with #407 or without it?

**Comment by ordian** [comment]
Your test works fine on a big-endian CPU as well (#508) with `uint` 0.9.0.

**Comment by That3Percent** [comment]
After investigation, I found that this panic happens only if `byteorder` gets installed at version `1.4.0`. If I depend on `primitive-types` in a new project this panic does not occur. But, for some reason `byteorder` was selected at `1.4.0` in the project that counts.

Here's a minimum repro:
```toml
[package]
name = "uint-panic"
version = "0.1.0"

[dependencies]
primitive-types = "0.8"
byteorder = "=1.4.0"
```

```rust
#[test]
fn panics() {
    use primitive_types::U256;
    let mut b32 = <[u8; 32]>::default();
    let a = U256::from(10);
    a.to_little_endian(&mut b32);
    let b = U256::from_little_endian(&b32[..]);
    assert_eq!(a, b);
}
```

To ensure this doesn't happen for other people I might recommend setting the minimum version of the `byteorder` dependency in your project to `1.4.1` or `1.4.2`.

**Comment by ordian** [comment]
Thanks for the investigation, looks like it was a regression in byteorder 1.4.0 that was fixed in 1.4.1: https://github.com/BurntSushi/byteorder/commit/0ead1057d4d1ea59ad9c8e5bc35514646ef8fb83.
